src/timer.c

```diff
@@ -275,7 +275,8 @@ static int gettime(int casep)
 #ifdef MAIKO_OS_HAIKU
```

Comment:
OK, but there's exactly ONE call to `settimeofday`, from `subr_settime()` also in timer.c, where it already has `#ifdef DOS ... #else...` to handle systems that have differing implementations for the time-of-day setting.  You can just eliminate all code except a `return;` in the Haiku case.  There's no checking of the return value - if it works, it works, otherwise nothing in the code cares.